docsrc/sasl/authentication_mechanisms.rst

```diff
@@ -10,35 +10,50 @@ Mechanisms
 ANONYMOUS
 ---------
 
-.. todo::
-   Content needed here
+As used in FTP this mechanism only passes an email address for authentication.
+
+Defined in :rfc:`2245`
+
+.. _MECH-CRAM-MD5:
 
 CRAM-MD5
 --------
 
-.. todo::
-   Content needed here
+This mechanism avoids sending the users password over the network in plain text by hashing the password with a server provided random value (known as a nonce).
+A disadvantage of this mechanism is that the server must maintain a database of **plaintext passwords** for comparison.
+
+Defined in :rfc:`2195`
+
+`draft-ietf-sasl-crammd5 <https://tools.ietf.org/html/draft-ietf-sasl-crammd5>`_ - The CRAM-MD5 SASL Mechanism (CRAM-MD5)
 
 
 DIGEST-MD5
 ----------
 
-.. todo::
-   Content needed here
+This mechanism improves upon the :ref:`MECH-CRAM-MD5` mechanism by avoiding the need for the server to store plaintext passwords.
+With digest authentication the server needs to store the **MD5 digest** of the users password which helps to make the system more secure.
+As in :ref:`MECH-CRAM-MD5` the password is hashed with a server nonce and other data before being transmitted across the network.
+
+Defined in :rfc:`2831`
 
 EXTERNAL
 --------
 
-.. todo::
-   Content needed here
+EXTERNAL is a SASL Mechanism allows a client to request the server to use credentials established by means external to the mechanism to authenticate the client.
 
+SASL EXTERNAL means may be, for instance, IP Security RFC 4301 or TLS services.
+In absence of some a prior agreement between the client and the server, the client cannot make any assumption as to what SASL EXTERNAL means the server has used to obtain the client's credentials, nor make an assumption as to the form of credentials.
```

Comment:
s/of some/of/

Reply:
Thanks, I will push all suggestions when your review is finished to avoid notification spam